### PR TITLE
pmemfile: wrappers for more syscalls

### PR DESCRIPTION
**Pull request by GBuella**
SYS_mmap
SYS_readv
SYS_writev
SYS_dup
SYS_dup2
SYS_dup3
SYS_sendfile
SYS_readlink
SYS_readlinkat
SYS_chroot
SYS_readahead
SYS_listxattr
SYS_llistxattr
SYS_flistxattr
SYS_removexattr
SYS_lremovexattr
SYS_fremovexattr
SYS_fadvise64
SYS_utime
SYS_utimes
SYS_futimesat
SYS_splice
SYS_utimensat
SYS_fallocate
SYS_name_to_handle_at
SYS_execve
SYS_execveat
SYS_copy_file_range
SYS_preadv2
SYS_pwritev2

---
This change is [Reviewable](https://reviewable.io/reviews/marcinslusarz/nvml/44)

### CODE REVIEW

File: src/libpmemfile/preload.c [hunk 1]
```diff
@@ -1636,3 +1780,153 @@ hook_fchownat(struct fd_desc at, const char *path,
...
+			long arg4, long arg5)
+{
+	struct resolved_path where;
+
+	resolve_path(cwd_desc(), (const char *)path, &where, resolve_last);
+
+	if (where.error_code != 0)
+		return where.error_code;
+
+	/* XXX */
```
Comment (marcinslusarz) [inline src/libpmemfile/preload.c]: Why? Please expand this comment.

Comment (GBuella) [reply src/libpmemfile/preload.c]: XXX

Comment (GBuella) [reply src/libpmemfile/preload.c]: It is my problem with path resolution, and _at interfaces ( or lack of them sometimes ).
I'll describe it in a comment, maybe tomorrow -- I need to find good way to explain it.

Comment (GBuella) [reply src/libpmemfile/preload.c]: I added some comments.

**Pull request merged**